File: backend/src/main/java/at/ac/tuwien/sepr/groupphase/backend/entity/JobOffer.java
package at.ac.tuwien.sepr.groupphase.backend.entity;

import at.ac.tuwien.sepr.groupphase.backend.type.JobOfferStatus;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

/**
 * Represents a job offer in the persistent data store.
 */
@Entity
@Table(name = "job_offer")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class JobOffer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "job_request_id", nullable = false)
    private JobRequest jobRequest;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "worker_id", nullable = false)
    private ApplicationUser worker;

    private float price;

    @Column(length = 1023)
    private String comment;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private JobOfferStatus status = JobOfferStatus.PENDING;

    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: frontend/src/app/dtos/jobOffer.ts
import {JobOfferStatus} from "./JobOfferStatus";

export interface JobOffer {
  id: number;
  comment?: string;
  createdAt: string;
  status: string;
  price: number;
  jobRequestId: number;
  workerId: number;
}

export interface JobOfferCreate {
  price: number;
  comment?: string;
}

export interface JobOfferSummary {
  id: number;
  price: number;
  status: JobOfferStatus;
  jobRequestId: number;
  requestTitle: string;
  lowestPrice: number;
}

export interface JobOfferWithWorker {
  id: number;
  comment?: string;
  createdAt: string;
  status: string;
  price: number;
  jobRequestId: number;
  workerId: number;
  workerUsername: string;
  workerAverageRating?: number;
}
